subsys/net/ip/tcp2.c

```diff
@@ -756,6 +751,7 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 	struct tcp *conn = NULL;
 	struct net_context *context = NULL;
 	sa_family_t af = net_pkt_family(pkt);
+	struct sockaddr local_addr = { 0 };
```

Comment:
How about to simplify this patch this way?

```
struct sockaddr *local_addr = net_sin_ptr(&context->local);
```

Reply:
sockaddr_in is not compatible with sockaddr_in_ptr (the address is pointer in latter in order to save some memory) so we need to copy

Reply:
What I meant, we can just get this pointer with net_sin_ptr() and use it, not?